learning_journal/tests.py

```diff
@@ -98,72 +78,195 @@ def test_home_list_returns_empty_when_empty(dummy_request):
+    from webtest import TestApp
+    from learning_journal import main
 
-# def test_home_route_has_table(testapp):
-#     """The home page has a table in the html."""
-#     response = testapp.get('/', status=200)
-#     html = response.html
-#     import pdb; pdb.set_trace()
-#     assert len(html.find_all("table")) == 1
+    app = main({}, **{'sqlalchemy.url': 'sqlite:///:memory:'})
+    testapp = TestApp(app)
 
+    session_factory = app.registry["dbsession_factory"]
+    engine = session_factory().bind
+    Base.metadata.drop_all(bind=engine)
+    Base.metadata.create_all(bind=engine)
 
-# def test_home_route_with_data_has_filled_table(testapp, fill_the_db):
-#     """When there's data in the database, the home page has some rows."""
-#     response = testapp.get('/', status=200)
-#     html = response.html
-#     assert len(html.find_all("tr")) == 101
+    return testapp
 
 
-# def test_home_route_has_table2(testapp):
-#     """Without data the home page only has the header row in its table."""
-#     response = testapp.get('/', status=200)
-#     html = response.html
-#     assert len(html.find_all("tr")) == 1
+@pytest.fixture
+def fill_the_db(testapp):
+    """Fill the database with some model instances."""
+    session_factory = testapp.app.registry["dbsession_factory"]
+    with transaction.manager:
+        dbsession = get_tm_session(session_factory, transaction.manager)
+        for entry in ENTRIES:
+            row = Entry(title=entry["title"],
+                        creation_date=entry["creation_date"],
+                        body=entry["body"])
+            dbsession.add(row)
+
+
+def test_home_route_does_not_have_all_lists(testapp):
+    """The home page has all html elements from index template."""
+    response = testapp.get('/', status=200)
+    html = response.html
+    assert len(html.find_all("main")) == 1
+    assert len(html.find_all("ul")) == 3
+
+
+def test_home_route_has_entrys(testapp, fill_the_db):
+    """Test that the home page has all listed entries."""
+    response = testapp.get('/', status=200)
+    html = response.html
+    assert html.find_all('li')[3].a.getText() == "It's Monday Dude"
+    assert html.find_all('li')[4].a.getText() == "It's Tuesday Dude"
+
+
+def test_new_entry_route_has_input_and_textarea(testapp):
+    """Test that new entry route has input and textarea."""
+    response = testapp.get('/journal/new-entry', status=200)
+    html = response.html
+    assert len(html.find_all("input")) == 2
+    assert len(html.find_all("textarea")) == 1
+
+
+def test_new_entry_route_creates_new_entry_in_db(testapp):
+    """Test that new entry route creates new entry in db."""
+    title = {
+        'title': 'I have a dream.',
+        'body': 'sup'
+    }
+    response = testapp.post('/journal/new-entry', title, status=302)
+    full_response = response.follow().html.find(class_='container')
+    assert full_response.li.a.text == title["title"]
```

Comment:
Can you use a db query object to test that your entry was added to the db.  use a different test to prove that if you add you can retrieve it.